recipe/meta.yaml

```
@@ -1,45 +1,52 @@
 {% set name = "paramiko" %}
-{% set version = "2.8.1" %}
+{% set version = "3.5.0" %}
 {% set bundle = "tar.gz" %}
 {% set hash_type = "sha256" %}
-{% set hash_val = "85b1245054e5d7592b9088cc6d08da22445417912d3a3e48138675c7a8616438" %}
+{% set hash_val = "ad11e540da4f55cedda52931f1a3f812a8238a7af7f62a60de538cd80bb28124" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.{{ bundle }}
   url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.{{ bundle }}
   {{ hash_type }}: {{ hash_val }}
 
 build:
+  skip: True  # [py<36]
   number: 0
-  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
-  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv --no-build-isolation
 
 requirements:
   host:
     - python
     - pip
+    - setuptools
+    - wheel
   run:
     - python
-    - bcrypt >=3.1.3
-    - cryptography >=2.5
-    - pynacl >=1.0.1
+    - bcrypt >=3.2
+    - cryptography >=3.3
+    - pynacl >=1.5
```

```suggestion
    - pynacl >=1.5
  run_constrained:
    - pyasn1 >=0.1.7
    - python-gssapi >=1.4.1  # [not win]
    - pywin32 >=218  # [win]
    - invoke >=2.0
```

Consider adding [optional dependencies](https://github.com/paramiko/paramiko/blob/23f92003898b060df0e2b8b1d889455264e63a3e/setup.py#L37-L43) to `run_constrained`

---

Good call.

---

Also, I've not seen the `pywin32`-style constraint before.